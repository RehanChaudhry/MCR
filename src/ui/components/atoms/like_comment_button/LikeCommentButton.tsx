import React, { useState } from "react";
import {
  ImageStyle,
  StyleProp,
  StyleSheet,
  TextStyle,
  TouchableOpacity,
  TouchableOpacityProps,
  View,
  ViewStyle
} from "react-native";
import { usePreferredTheme } from "hooks";
import { AppLabel } from "ui/components/atoms/app_label/AppLabel";
import { FONT_SIZE, FONTS } from "config";
import { Color } from "react-native-svg";

export interface LikeButtonProps extends TouchableOpacityProps {
  onValueChanged?: (isSelected: boolean) => void;
  buttonStyle?: StyleProp<ViewStyle>;
  textStyle?: StyleProp<TextStyle>;
  selectedText?: string;
  unSelectedText: string;
  iconStyle?: StyleProp<ImageStyle>;
  icon?:
    | ((isSelected: boolean, color: Color) => React.ReactElement)
    | null;
}

export const LikeCommentButton = React.memo<LikeButtonProps>(
  ({
    onValueChanged,
    buttonStyle,
    textStyle,
    icon,
    selectedText,
    unSelectedText
  }) => {
    const theme = usePreferredTheme();
    const [isSelected, setIsSelected] = useState(false);

    return (
      <TouchableOpacity
        onPress={() => {
          setIsSelected(!isSelected);
          onValueChanged?.(!isSelected);
        }}
        style={[
          style.button,
          {
            backgroundColor: theme.themedColors.tertiaryIconColor
          },
          buttonStyle
        ]}>
        <View testID="button-container" style={style.viewContainer}>
          {icon?.(isSelected, theme.themedColors.primaryIconColor)}
          <AppLabel
            style={[
              style.text,

              isSelected
                ? textStyle
                : { color: theme.themedColors.primaryLabelColor }
            ]}
            text={
              (isSelected ? selectedText : unSelectedText) ??
              unSelectedText
            }
            weight={"semi-bold"}
          />
        </View>
      </TouchableOpacity>
    );
  }
);

const style = StyleSheet.create({
  button: {
    flexDirection: "row",
    paddingVertical: 10,
    justifyContent: "center",
    alignItems: "center",
    borderRadius: 5,
    flex: 0,
    width: "auto",
    paddingHorizontal: 16,
    alignSelf: "flex-start"
  },
  viewContainer: {
    flexDirection: "row",
    alignItems: "center"
  },
  leftIcon: {
    width: 10,
    height: 10,
    resizeMode: "contain"
  },
  text: {
    fontSize: FONT_SIZE.sm,
    marginStart: 5,
    fontFamily: FONTS.semiBold
  }
});
