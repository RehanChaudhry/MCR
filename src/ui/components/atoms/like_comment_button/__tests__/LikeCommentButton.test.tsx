import { fireEvent, render } from "@testing-library/react-native";
import React from "react";
import { LikeCommentButton } from "ui/components/atoms/like_comment_button/LikeCommentButton";
import Like from "assets/images/like.svg";

test("snapshot testing", () => {
  const rendered = render(
    <LikeCommentButton unSelectedText={"Like"} />
  ).toJSON();
  expect(rendered).toMatchSnapshot();
});

test("should properly perform click event", () => {
  const onPress = jest.fn();
  const { getByText } = render(
    <LikeCommentButton unSelectedText={"Like"} onValueChanged={onPress} />
  );
  const button = getByText("Like");
  fireEvent.press(button);
  expect(onPress).toBeCalledTimes(1);
});

it("renders  icon", () => {
  const { queryByTestId } = render(
    <LikeCommentButton
      unSelectedText={"Like"}
      icon={() => <Like testID={"icon"} width={16} height={16} />}
    />
  );
  const icon = queryByTestId("icon");
  expect(icon).not.toBeNull();
});
