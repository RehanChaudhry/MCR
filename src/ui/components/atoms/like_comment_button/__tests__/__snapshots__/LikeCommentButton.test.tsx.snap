// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "flex-start",
      "backgroundColor": "#3C3C434D",
      "borderRadius": 5,
      "flex": 0,
      "flexDirection": "row",
      "justifyContent": "center",
      "opacity": 1,
      "paddingHorizontal": 16,
      "paddingVertical": 10,
      "width": "auto",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
    testID="button-container"
  >
    <Text
      numberOfLines={1}
      style={
        Array [
          Array [
            Object {
              "fontFamily": "Inter-Regular",
            },
            Object {
              "color": "#000000",
            },
            Array [
              Object {
                "fontFamily": "Inter-SemiBold",
                "fontSize": 18.139805825242718,
                "marginStart": 5,
              },
              Object {
                "color": "#000000",
              },
            ],
          ],
          Object {
            "fontFamily": "Inter-SemiBold",
          },
        ]
      }
    >
      Like
    </Text>
  </View>
</View>
`;
